vignettes/intro_opencage.Rmd.src

```diff
@@ -0,0 +1,188 @@
+The "Free Trial" plan provides up to 2,500 API requests a day. 
+There are paid plans available, if you need to run more API requests.
+After you have registered, you can generate an API key at the [OpenCage dashboard](https://opencagedata.com/dashboard#api-keys).
+
+Now, you will need to pass the API key on to the functions in {opencage}.
+They will conveniently retrieve your API key if it is saved in the environment variable `"OPENCAGE_KEY"`. 
+If it is not, `oc_config()` will help to set that environment variable.
+
+Do not pass the key directly as a parameter to the function, though, as you risk exposing it via your script or your history. 
+There are three safer ways to set your API key instead:
+
+1. Save your API key as an environment variable in `.Renviron` as described in [What They Forgot to Teach You About R](https://rstats.wtf/r-startup.html#renviron) or [Efficient R Programming](https://csgillespie.github.io/efficientR/set-up.html#renviron).
+From there it will be fetched by all functions that call the OpenCage API, so you do not even have to call `oc_config()` to set your key, but can start geocoding right away. 
+If you have the {[usethis](https://usethis.r-lib.org/)} package installed, you can edit your `.Renviron` most easily with `usethis::edit_r_environ()`.
+We strongly recommend storing your API key in the user-level `.Renviron`, as opposed to the project-level, because this makes it less likely you will share sensitive information by mistake.
+
+2. If you use a package like {[keyring](https://github.com/r-lib/keyring)} to store your credentials, you can safely pass your key in a script with a function call like this `oc_config(key = keyring::key_get("opencage"))`.
+
+3. If you call `oc_config()` in an `interactive()` session and the `OPENCAGE_KEY` environment variable is not set, it will prompt you to enter the key in the console.
+
+Whatever method you choose, keep your API key secret. OpenCage also features [best practices for keeping your API key safe](https://opencagedata.com/protecting-your-api-key). 
+
+### Rate limit
+
+A rate limit is used to control the rate of requests sent, so legitimate requests do not lead to an unintended Denial of Service attack.
+The rate limit allowed by the API depends on the OpenCage plan you purchased and ranges from 1 request/sec for the "Free Trial" plan to 15 requests/sec for the "Medium" or "Large" plans. 
+See [opencagedata.com/pricing](https://opencagedata.com/pricing) for details and up-to-date information. 
+
+If you have a "Free Trial" account with OpenCage, you can skip to the next section, because the rate limit is already set correctly for you at 1 request/sec.  
+
+If you have a paid account, you can set the rate limit for the active R session with `oc_config(rate_sec = n)` (where `n` is the appropriate rate limit). 
+You can set the rate limit persistently across sessions by setting an `oc_rate_sec` option in your `.Rprofile`. 
```

Comment:
add that the Rprofile can be specific to an R project?

Reply:
In order to have differing rate limits (per project) you would have to have multiple paid plans. My guess is that this is not very common. This is also explained in more detail in WTF for example, which we link to.
But maybe I just see it this way, because I hardly use project-specific `.Rprofile`s. Do you use them?

Reply:
I don't and you're right, curious people would dig a bit further.